ui/imports/shared/status/StatusImageModal.qml

```diff
@@ -31,39 +32,81 @@ StatusDialog {
 
         property int maxHeight: Global.applicationWindow.height - 80
         property int maxWidth: Global.applicationWindow.width - 80
+        readonly property int radius: Style.current.radius
     }
 
-    onOpened: {
-        messageImage.source = root.image.source;
-    }
+    onOpened: imageLoader.source = root.image.source;
+    onClosed: imageLoader.source = ""
+
+    contentItem: Loader {
+        id: imageLoader
+
+        readonly property bool isError: status === Loader.Error || (imageLoader.item && imageLoader.item.status === Image.Error)
+        readonly property bool isLoading: status === Loader.Loading || (imageLoader.item && imageLoader.item.status === Image.Loading)
+        property string source
 
-    contentItem: AnimatedImage {
-        id: messageImage
         anchors.fill: parent
-        asynchronous: true
-        fillMode: Image.PreserveAspectFit
-        mipmap: true
-        smooth: false
+        active: true
+        sourceComponent: root.plain ? plainImage : animatedImage
 
-        onStatusChanged: playing = (status == AnimatedImage.Ready)
         MouseArea {
             anchors.fill: parent
             acceptedButtons: Qt.LeftButton | Qt.RightButton
             onClicked: {
                 if (mouse.button === Qt.LeftButton)
                     root.close()
-                if (mouse.button === Qt.RightButton)
-                    Global.openMenu(imageContextMenu,
-                                    messageImage,
-                                    { imageSource: messageImage.source, url: root.url})
+                if (imageLoader.isError || imageLoader.isLoading || mouse.button !== Qt.RightButton)
+                    return
+                const isGif = (!root.plain && imageLoader.item && imageLoader.item.playing) || imageLoader.source.toLowerCase().endsWith(".gif")
```

Comment:
hmm I don't think we should rely on the file extension to determine the media type.